python/lsst/meas/extensions/ngmix/LMSimpleShape.py

```diff
@@ -0,0 +1,238 @@
+        ("flag_singularMatrix", "Singular Matrix"),
+        ("flag_negativeCovEig", "Negative Covariance Eig"),
+        ("flag_negativeCoveDiag", "Negative Covariance Diag"),
+        ("flag_eigNotFinite", "Eig not finite"),
+        ("flag_funcNotFinite", "Func not finite"),
+        ("flag_divideByZero", "Divide by Zero"))
+class SingleFrameNgmixPlugin(SingleFramePlugin):
+    '''
+    Plugin to do shape measurement using ngmix.LMSimple fitter
+    May be run using gauss, dev, or exp model
+    '''
+    ConfigClass = SingleFrameNgmixConfig
+
+    #   Length of a single gaussian representation in ngmix
+    gaussian_pars_len = 6
+
+    @classmethod
+    def getExecutionOrder(cls):
+        return 3.0
+
+    #   This defines the legal LMSimple routines to call
+    #   This metadata will be fetched from Erin when he adds an API for it
+    def getModelInfo(self, model):
+        modelInfo = {"gauss": {"nPars": 6, "nGauss": 1},
+                     "dev": {"nPars": 6, "nGauss": 6},
+                     "exp": {"nPars": 6, "nGauss": 6}}
+        if model not in modelInfo.keys():
+            raise lsst.pex.exceptions.RuntimeError("Illegal model: %s for LMSimple"%model)
+        return modelInfo[model]
+
+    def __init__(self, config, name, schema, metadata):
+        SingleFramePlugin.__init__(self, config, name, schema, metadata)
+
+        # Add ShapeletFunction keys for the number of Gaussians required
+        self.keys = []
+        self.nGauss = self.getModelInfo(self.config.model)["nGauss"]
+        #   Add a 0th order shapelet for every Gaussian to be returned
+        for i in range(self.nGauss):
+            key = ShapeletFunctionKey.addFields(schema, "%s_%d"%(name, i), "ngmix Shape gaussian",
+                                                "pixels", "", 0, HERMITE)
+            self.keys.append(key)
+        #   If the psfName is specified, it means to we can get the psfApprox from it
+        self.msfKey = MultiShapeletFunctionKey(self.keys)
+        if not self.config.psfName is None:
+            self.psfMsfKey = MultiShapeletFunctionKey(schema[self.config.psfName], HERMITE)
+        else:
+            self.psfMsfKey = None
+        self.name = name
+
+    def measure(self, measRecord, exposure):
+        psf = exposure.getPsf()
+        if psf is None:
+            raise MeasurementError(self.flagHandler.getDefinition(
+                                   SingleFrameNgmixPlugin.ErrEnum.flag_noPsf).doc,
+                                   SingleFrameNgmixPlugin.ErrEnum.flag_noPsf)
+        # make an observation for the psf image
+        psfArray = psf.computeKernelImage().getArray()
+        psfJacob = ngmix.UnitJacobian(row=(psfArray.shape[0] - 1)/2, col=(psfArray.shape[1] - 1)/2)
+        psfObs = Observation(psfArray, jacobian=psfJacob)
+
+        #   Fallback code if no psf algorithm is requested.  Then just do an LM single gaussian fit
+        if self.config.psfName is None:
```

Comment:
I am not sure the idea of having a string which is defined only as a field name prefix be used as a control flow mechanism. How does the user know to get this behavior without reading the source code? I think it may be better to have some explicitly config variable for this control flow, or somehow expose this to the end user better